src/cross_toml.rs

```diff
@@ -89,6 +90,52 @@ impl CrossToml {
         Ok((cfg, unused))
     }
 
+    /// Merges another [`CrossToml`] into `self` and returns a new merged one
+    ///
+    /// # Merging of `targets`
+    /// The `targets` entries are merged based on the [`Target`] keys.
+    /// If a [`Target`] key is present in both configs, the [`CrossTargetConfig`]
+    /// in `other` overwrites the one in `self`.
+    ///
+    /// # Merging of `build`
+    /// The `build` fields ([`CrossBuildConfig`]) are merged based on their sub-fields.
+    /// A field in the [`CrossBuildConfig`] will only overwrite another if it contains
+    /// a value, i.e. it is not `None`.
+    pub fn merge(&self, other: &CrossToml) -> Result<CrossToml> {
```

Comment:
Couldn't this be implemented before deserializing the two/three data sources? 

e.g take the file contents, make `toml::Table`s, extend them, and then deserialize back.

If we serialize like this we lose unused keys, losing some information when we report back, ofc, this isn't a problem because we reported it already when making